### PR TITLE
Add terminal command, moves completion and terminal commands to utils

### PR DESCRIPTION
**Pull request by cdrage**
This combines both autocompletion and "terminal" commands into one
category (`odo utils`).

The new command added is `odo utils terminal`

This can be used with either bash/zsh by: source <(odo utils terminal
bash/zsh)

### CODE REVIEW

**Comment by cdrage** [comment]
This removes `odo completion` in favour of `odo utils completion` and `odo utils terminal` @kadel

**Comment by surajnarwade** [comment]
```
$ odo app create myapp
Creating application: myapp
Switched to application: myapp
odo: There's no active application.
You can create one by running 'odo application create <name>'. - No component is set as current
[snarwade@redsuraj ~]$ 
odo: There's no active application.
You can create one by running 'odo application create <name>'. - No component is set as current
[snarwade@redsuraj ~]$ 
odo: There's no active application.
You can create one by running 'odo application create <name>'. - No component is set as current
[snarwade@redsuraj ~]$ odo app list
ACTIVE   NAME
  *      myapp
odo: There's no active application.
You can create one by running 'odo application create <name>'. - No component is set as current
```

I have active application, still it shows no active application

**Comment by kadel** [comment]
I'm having the same problem as Suraj
```
▶ odo create wildfly
Please wait, creating wildfly component ...
Component 'wildfly' was created.
To push source code to the component run 'odo push'

Component 'wildfly' is now set as active component.
odo: app - No component is set as current
▶
```

**Comment by cdrage** [comment]
@kadel @surajnarwade 

Can you make sure that your built Odo binary is located at $GOPATH ? (sudo cp odo $GOPATH/bin/odo).

The function uses just `odo` not `./odo` so it'll be where the binary could be found.

**Comment by kadel** [comment]
@cdrage you mean $PATH, not $GOPATH, right? It should expect that `odo` is in $PATH.

The problem is that it doesn't get updated when you change components.  

```
odo: app - wildfly
▶ odo list
ACTIVE     NAME                  TYPE
           another-component     wildfly
*          wildfly               wildfly

odo: app - wildfly
▶ odo component
The current component is: wildfly

odo: app - wildfly
▶ odo component another-component
Switched to component: another-component

odo: app - wildfly
▶ odo list
ACTIVE     NAME                  TYPE
*          another-component     wildfly
           wildfly               wildfly

odo: app - wildfly
▶ odo component
The current component is: another-component

odo: app - wildfly
▶
```

**Comment by cdrage** [comment]
@surajnarwade Updated with your suggestion! Ready for another review :+1:

**Comment by cdrage** [comment]
@kadel @surajnarwade 

It should change now, I had to switch the command to using single quotes instead of double, as the function (should) be called each time you press enter on your terminal.

Please feel free to test it out!

**Comment by kadel** [comment]
It works 👍

File: cmd/cmd_utils.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
+package cmd
```
Comment (kadel) [inline cmd/cmd_utils.go]: shouldn't be this file called `cmdutils.go`?
Underscore in go source file names are usually used to separate tests or architectures `_linux.go`, `_win.go` ...
It looks a bit strange here, but maybe it is just me ;-)

**Comment by cdrage** [comment]
@kadel Renamed the file! :+1: Merge time?

**Pull request merged**